src/components/Configure/content/useMutateInstallation.tsx

```diff
@@ -0,0 +1,47 @@
+import { useApiKey } from '../../../context/ApiKeyProvider';
+import { useConnections } from '../../../context/ConnectionsContext';
+import { useErrorState } from '../../../context/ErrorContextProvider';
+import { useInstallIntegrationProps } from '../../../context/InstallIntegrationContext';
+import { useProject } from '../../../context/ProjectContext';
+import { useSelectedObjectName } from '../ObjectManagementNav';
+import { useConfigureState } from '../state/ConfigurationStateProvider';
+import { useHydratedRevision } from '../state/HydratedRevisionContext';
+import { getConfigureState } from '../state/utils';
+
+/**
+ * state hook for installation flows (Create/Update)
+ * */
+export const useMutateInstallation = () => {
+  const {
+    integrationId, groupRef, consumerRef, setInstallation,
+  } = useInstallIntegrationProps();
+  const { hydratedRevision, loading } = useHydratedRevision();
+  const { selectedObjectName } = useSelectedObjectName();
+  const { selectedConnection } = useConnections();
+  const apiKey = useApiKey();
+  const { projectId } = useProject();
+  const { resetBoundary, setErrors } = useErrorState();
+  const {
+    resetConfigureState, objectConfigurationsState, resetPendingConfigurationState,
+  } = useConfigureState();
+  const configureState = getConfigureState(selectedObjectName || '', objectConfigurationsState);
+
+  return {
+    integrationId,
+    groupRef,
+    consumerRef,
+    setInstallation,
+    hydratedRevision,
+    loading,
+    selectedObjectName,
+    selectedConnection,
+    apiKey,
```

Comment:
(Not in this PR, but in the future), it would be good to introduce a `useAPI` hook [just like in the console](https://github.com/amp-labs/client/blob/main/src/data/api.ts) (but the useAPI hook in this codebase would inject the API key, not the token like in the Console.)

Reply:
Yeah makes sense!